### ISSUE TITLE
Minishift can't download 3.11.0 oc binary

### ISSUE THREAD

**Issue by ankitverma123** [minishift/minishift#3378]
Not able to download oc binary. Below error it's throwing.

C:\minishift-1.34.1-windows-amd64>minishift.exe start --vm-driver virtualbox
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.11.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.11.0' is supported ... OK
-- Checking if requested hypervisor 'virtualbox' is supported on this platform ... OK
-- Checking if VirtualBox is installed ... OK
-- Checking the ISO URL ... OK
-- Downloading OpenShift binary 'oc' version 'v3.11.0'
-- Downloading OpenShift v3.11.0 checksums ... OK**Error starting the cluster: Error attempting to download and cache 'oc': Failed to validate hash - expected: a802eb5364fcfda76dd297c2138409fcb0e6dca60ed4cecfa2fc5bfe16a20732, actual: cdb84cc0000d0f0983120f903b2cad7114527ce2a9c4eb1988986eda7b877bfa**

**Comment by ankitverma123** [comment]
@mik-dass  could you please look into this.

**Comment by anjannath** [comment]
@ankitverma123 Is this still an issue for you, am not able to reproduce it. maybe the file got corrupted while downloading..
```
-- Checking if Libvirt default network is present ... OK
-- Checking if Libvirt default network is active ... OK
-- Checking the ISO URL ... OK
-- Downloading OpenShift binary 'oc' version 'v3.11.0'
 53.89 MiB / 53.89 MiB [==========================================================================================================================================================] 100.00% 0s
-- Downloading OpenShift v3.11.0 checksums ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'kvm' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    4 GB
```

**Comment by ankitverma123** [comment]
@anjannath  Yes i just downloaded the latest version of minishift 1.34.1 https://github.com/minishift/minishift/releases/tag/v1.34.1
and tried using this but same results. 

C:\minishift>minishift start --openshift-version 3.10.0
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.10.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.10.0' is supported ... OK
-- Checking if requested hypervisor 'virtualbox' is supported on this platform ... OK
-- Checking if VirtualBox is installed ... OK
-- Checking the ISO URL ... OK
-- Downloading OpenShift binary 'oc' version 'v3.10.0'
-- Downloading OpenShift v3.10.0 checksums ... OKError starting the cluster: Error attempting to download and cache 'oc': Failed to validate hash - expected: 46ae5ccccc7f774ba5b0d8e1621a428e2c3fae7b7c0c78872cb1b37e4495b2e3, actual: c1b33aa535b88898d0622e0af2aa673bb814c354fb438c21c18155afc51acf87



I also found one suspicious thing https://github.com/minishift/minishift/blob/master/pkg/util/github/github.go#L162 
at this line of code for the release.

**Comment by anjannath** [comment]
> I also found one suspicious thing https://github.com/minishift/minishift/blob/master/pkg/util/github/github.go#L162

@ankitverma123 What is suspicious about it?

since am not able to reproduce this, as a workaround you can manually download the `oc` binary and put it in the location minishift expects to find it. which is `~/.minishift/cache/oc/v3.11.0/windows/oc`

**Comment by bellhop** [comment]
I am seeing the same issue on mac, can you varify the path to the cache for all three os?

Below you can see that the actual hash is correct based on the info listed in the release but the expected value is not correct.

---
```
minishift start
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.11.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.11.0' is supported ... OK
-- Checking if requested hypervisor 'xhyve' is supported on this platform ... OK
-- Checking if xhyve driver is installed ...
   Driver is available at /usr/local/bin/docker-machine-driver-xhyve
   Checking for setuid bit ... OK
-- Checking the ISO URL ... OK
-- Downloading OpenShift binary 'oc' version 'v3.11.0'
-- Downloading OpenShift v3.11.0 checksums ... OKError starting the cluster: Error attempting to download and cache 'oc': Failed to validate hash - expected: 252be97fc39eed1002fb509e2d25217718498b8bc8b77f5788e090aa6847fa3c, actual: 75d58500aec1a2cee9473dfa826c81199669dbc0f49806e31a13626b5e4cfcf0
```

**Comment by Mert84** [comment]
I've observed this problem on my windows 10. 
The script was idling during the download, showing only 16KB in the download status.
I've begun creating directory tree ~/.minishift/cache/oc/v3.11.0/windows/oc  anjannath had suggested.
The moment I've created directory named "v3.11.0" the rest of the directories have been generated by themselves. I switched to the command prompt and seen that download was going smoothly after that. 
A wild guess would be : the script failed to create the "v3.11.0" folder and somehow failed to catch the error and was waiting until timeout.
